splash/qtrender_lua.py

```diff
@@ -452,7 +452,21 @@ def set_custom_headers(self, headers):
     def set_viewport(self, size):
         if size is None:
             return
-        return self.tab.set_viewport(size)
+        return self.tab.set_viewport(size, raise_if_empty=True)
+
+    @command()
+    def get_viewport_size(self):
+        sz = self.tab.web_page.viewportSize()
+        return sz.width(), sz.height()
+
+    @command()
+    def get_window_size(self):
+        sz = self.tab.web_view.size()
+        return sz.width(), sz.height()
+
+    @command()
+    def set_window_size(self, sz):
+        self.tab.set_window_size(sz)
```

Comment:
I see 3 options:

1. "WxH" strings. Strings make it harder to get or change individual size components, I don't like them either.
2. Multiple return values: `local w, h = splash:get_window_size()`, then `splash:set_window_size(100, 200)` and `splash:set_window_size{width=w, height=h}`. It seems that `splash:set_window_size(splash:get_window_size())` still works here.
3. A table: `local sz = splash:get_window_size()`, then `splash:set_window_size{width=sz.width, height=sz.height}` or `splash:set_window_size(sz)`. The return value is a Lua table here, not a QSize object. 

(3) doesn't prevent `splash:set_window_size(100, 200)` syntax from being supported. So the difference between (2) and (3) boils down to the return value of `splash:get_window_size()`. I like (3) a bit more because it makes it harder to accidentally discard a size component, and because it allows for easier/more consistent error reporting if we need it (return nil, error). But (2) is also ok because it doesn't look like we need error reporting here (do we?).